jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/ResponseHeaderRegexRule.java

```diff
@@ -0,0 +1,138 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.rewrite.handler;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.StringUtil;
+
+/**
+ * Sends the response code whenever the rule finds a match.
+ */
+public class ResponseHeaderRegexRule extends Rule
+{
+    private String _headerName;
+    private Pattern _headerRegex;
+    private int _code;
```

Comment:
I didn't really look anywhere else than the existing rewrite modules, both of the existing ones that update the return status code used code:

InvalidURIRule.java:    private int _code = HttpStatus.BAD_REQUEST_400;
ResponsePatternRule.java:    private int _code;

I can change this, but then it will be configured differently than the existing modules which take a status code as a parameter? They will be "<Set name="code">403</Set>" and then this will be "<Set name="status">403</Set>".

Reply:
An obvious doh! here 😄